experiment/logviewer/gcsreader/main.go

```
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+)
+
+func main() {
+	defer timeTrack(time.Now(), "total run")
+	defer durations.printAll()
+	run()
+}
+
+func run() {
+	logPath, targetSubstring := setupFromConsole()
+	regex, _ := regexp.Compile(targetSubstring)
+
+	r, err := downloadAndDecompress(logPath)
+	handle(err)
+	parsed, _ := processLines(r, regex)
```

Ignoring an error here is probably bad idea.